### PR TITLE
Add rule S1539 (StrictMode) for JS/TS (new rule for TS)

### LINKED ISSUE #1805: Add rule S1539 (StrictMode) for JS/TS

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**
Fixes #1805

### CODE REVIEW

File: its/ruling/src/test/expected/js/p5.js/javascript-S1539.json [hunk 1]
```diff
@@ -5,17 +5,8 @@
 4625,
 ],
 'p5.js:lib/addons/p5.sound.js':[
-77,
```
Comment (quentin-jaquier-sonarsource) [inline its/ruling/src/test/expected/js/p5.js/javascript-S1539.json]: In fact, these missing issues are not due to the single quotes, but to the fact that the 'use strict' is not the first statement in the function, therefore this is not a correct 'use strict'.

File: its/ruling/src/test/expected/js/jshint/javascript-S1539.json [hunk 2]
```diff
@@ -98,16 +98,8 @@
 'jshint:tests/unit/fixtures/latedefundef.js':[
 16,
 ],
-'jshint:tests/unit/fixtures/reserved.js':[
```
Comment (vilchik-elena) [inline its/ruling/src/test/expected/js/jshint/javascript-S1539.json]: do you know why these issues are removed?

Comment (vilchik-elena) [reply its/ruling/src/test/expected/js/jshint/javascript-S1539.json]: parsing is failing, that's why

Comment (quentin-jaquier-sonarsource) [reply its/ruling/src/test/expected/js/jshint/javascript-S1539.json]: Yes, I confirm that reserved.js, strings.js and with.js parsing is failing. For strict_incorrect.js, it is due to the fact that the literal is not in first position.

**Pull request merged**